src/beliefpropagation/beliefpropagation_schedule.jl

```diff
@@ -1,25 +1,24 @@
-abstract type EdgeSequenceAlgorithm end
-
-struct ForestCover <: EdgeSequenceAlgorithm end
-struct Parallel <: EdgeSequenceAlgorithm end
-
-default_edge_sequence_alg() = ForestCover()
+default_edge_sequence_alg() = Algorithm("ForestCover")
 
 default_bp_niters(g::NamedGraph) = is_tree(g) ? 1 : nothing
-function default_bp_niters(g::Union{DataGraph,AbstractITensorNetwork})
+function default_bp_niters(g::AbstractGraph)
   return default_bp_niters(undirected_graph(underlying_graph(g)))
 end
 
 function edge_sequence(g::NamedGraph; alg=default_edge_sequence_alg())
   return edge_sequence(alg, g)
 end
 
-function edge_sequence(mts::DataGraph; alg=default_edge_sequence_alg())
-  return edge_sequence(alg, undirected_graph(underlying_graph(mts)))
+function edge_sequence(g::AbstractGraph; alg=default_edge_sequence_alg())
+  return edge_sequence(alg, undirected_graph(underlying_graph(g)))
```

Comment:
```suggestion
  return edge_sequence(Algorithm(alg), undirected_graph(underlying_graph(g)))
```
This change will allow you to just use string names not wrapped in `Algorithm` in other functions.